open-english-wordnet: init at 2022

nicoonoclaste opened the pull request:
## Description of changes

- Packaged [Open English WordNet](https://en-word.net/)
- Wrote a patch for reproducibility issues, submitted it upstream.

Having WordNet in nixpkgs would allow improving tools like `wordbook` which use that data, but currently download it at runtime (in the user's homedir)


## Things done

- Built on platform(s)
  - [x] x86_64-linux
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

nicoonoclaste committed:
open-english-wordnet: init at 2022

nicoonoclaste committed:
open-english-wordnet: Fix merge.py

- Upstream left the wrong version number in the script as tagged for the release
- The output depends on file iteration order.

FliegendeWurst commented on pkgs/by-name/op/open-english-wordnet/package.nix:
```diff
@@ -0,0 +1,73 @@
+, gzip
+, python3
+, stdenvNoCC
+, ... }:
```
```suggestion
}:
```

FliegendeWurst commented on pkgs/by-name/op/open-english-wordnet/package.nix:
```diff
@@ -0,0 +1,73 @@
+  '';
+
+  meta = with lib; {
+    description = "The Open English WordNet";
```
```suggestion
    description = "Lexical network of the English language";
```

nicoonoclaste committed:
open-english-wordnet: Use unique filename under `share/`

Otherwise users may have issues trying to bring multiple wordnets in their environment.

FliegendeWurst commented on pkgs/by-name/op/open-english-wordnet/package.nix:
```diff
@@ -0,0 +1,73 @@
+    "6da46a48dd76a48ad9ff563e6c807b8271fc83cd" = "sha256-QkkJH7NVGy/IbeSWkotU80IGF4esz0b8mIL9soHdQtQ=";
+  };
+
+  # TODO(nicoo): make compression optional?
```
Depends on what potential package users expect, I'd say.

FliegendeWurst reviewed (approved):
LGTM.

Result of `nixpkgs-review pr 265692` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>open-english-wordnet</li>
  </ul>
</details>
